cpp/tests/strings/like_tests.cpp

```diff
@@ -0,0 +1,161 @@
+  cudf::test::fixed_width_column_wrapper<bool> expected(
+    {true, false, false, false, false, false, false}, {1, 1, 1, 1, 1, 1, 0});
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+}
+
+TEST_F(StringsLikeTests, Leading)
+{
+  cudf::test::strings_column_wrapper input({"a", "aa", "aaa", "b", "bb", "bba", ""});
+  auto sv = cudf::strings_column_view(input);
+  {
+    auto results = cudf::strings::like(sv, std::string("a%"));
+    cudf::test::fixed_width_column_wrapper<bool> expected(
+      {true, true, true, false, false, false, false});
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+  }
+  {
+    auto results = cudf::strings::like(sv, std::string("__a%"));
+    cudf::test::fixed_width_column_wrapper<bool> expected(
+      {false, false, true, false, false, true, false});
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+  }
+}
+
+TEST_F(StringsLikeTests, Trailing)
+{
+  cudf::test::strings_column_wrapper input({"a", "aa", "aaa", "b", "bb", "bba", ""});
+  auto sv = cudf::strings_column_view(input);
+  {
+    auto results = cudf::strings::like(sv, std::string("%a"));
+    cudf::test::fixed_width_column_wrapper<bool> expected(
+      {true, true, true, false, false, true, false});
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+    results = cudf::strings::like(sv, std::string("%a%"));
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+  }
+  {
+    auto results = cudf::strings::like(sv, std::string("%_a"));
+    cudf::test::fixed_width_column_wrapper<bool> expected(
+      {false, true, true, false, false, true, false});
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+  }
+}
+
+TEST_F(StringsLikeTests, Place)
+{
+  cudf::test::strings_column_wrapper input({"a", "aa", "aaa", "bab", "ab", "aba", ""});
+  auto sv = cudf::strings_column_view(input);
+  {
+    auto results = cudf::strings::like(sv, std::string("a_"));
+    cudf::test::fixed_width_column_wrapper<bool> expected(
+      {false, true, false, false, true, false, false});
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+  }
+  {
+    auto results = cudf::strings::like(sv, std::string("_a_"));
+    cudf::test::fixed_width_column_wrapper<bool> expected(
+      {false, false, true, true, false, false, false});
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+  }
+  {
+    auto results = cudf::strings::like(sv, std::string("__a"));
+    cudf::test::fixed_width_column_wrapper<bool> expected(
+      {false, false, true, false, false, true, false});
+    CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(results->view(), expected);
+  }
+}
+
+TEST_F(StringsLikeTests, Escape)
+{
+  cudf::test::strings_column_wrapper input(
+    {"10%-20%", "10-20", "10%%-20%", "a_b", "b_a", "___", "", "axb"});
+  auto sv = cudf::strings_column_view(input);
+  {
+    auto results = cudf::strings::like(sv, std::string("10\\%-20\\%"), std::string("\\"));
```

Comment:
The test would be clearer if the pattern/escape were explicitly declared. Also applies to other tests with multiple arguments. You could sprinkle some more `const` around this test file.
```suggestion
    auto const pattern = std::string("10\\%-20\\%");
    auto const escape  = std::string("\\");
    auto const results = cudf::strings::like(sv, pattern, escape);
```